frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/useFeatureToggleSwitch.tsx

```diff
@@ -0,0 +1,235 @@
+                    return next();
+                }
+
+                setChangeRequestDialogCallback(() => {
+                    setChangeRequestDialogCallback(undefined);
+                    // always reset to previous state when using change requests
+                    config.onRollback?.();
+                });
+
+                onChangeRequestToggle(
+                    config.featureId,
+                    config.environmentName,
+                    newState,
+                    shouldActivateDisabledStrategies,
+                );
+            };
+
+            const handleToggleEnvironmentOn: Middleware = async (next) => {
+                if (newState !== true) {
+                    return next();
+                }
+
+                try {
+                    await toggleFeatureEnvironmentOn(
+                        config.projectId,
+                        config.featureId,
+                        config.environmentName,
+                        shouldActivateDisabledStrategies,
+                    );
+                    setToastData({
+                        type: 'success',
+                        title: `Enabled in ${config.environmentName}`,
+                        text: `${config.featureId} is now available in ${config.environmentName} based on its defined strategies.`,
+                    });
+                    config.onSuccess?.();
+                } catch (error: unknown) {
+                    setToastApiError(formatUnknownError(error));
+                    config.onRollback?.();
+                }
+            };
+
+            const handleToggleEnvironmentOff: Middleware = async (next) => {
+                if (newState !== false) {
+                    return next();
+                }
+
+                try {
+                    await toggleFeatureEnvironmentOff(
+                        config.projectId,
+                        config.featureId,
+                        config.environmentName,
+                    );
+                    setToastData({
+                        type: 'success',
+                        title: `Disabled in ${config.environmentName}`,
+                        text: `${config.featureId} is unavailable in ${config.environmentName} and its strategies will no longer have any effect.`,
+                    });
+                    config.onSuccess?.();
+                } catch (error: unknown) {
+                    setToastApiError(formatUnknownError(error));
+                    config.onRollback?.();
+                }
+            };
+
+            return composeAndRunMiddlewares([
+                confirmProductionChanges,
+                ensureActiveStrategies,
+                addToChangeRequest,
+                handleToggleEnvironmentOff,
+                handleToggleEnvironmentOn,
+            ]);
+        },
+        [setProdGuardModalState],
+    );
+
+    const modals = (
+        <>
+            <FeatureStrategyProdGuard {...prodGuardModalState} />
```

Comment:
The checked state of the `Don't show again` checkbox seems to be broken

Reply:
fixed